pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -108,4 +117,27 @@ func (r *ReconcileCluster) newStatus(cluster *synv1alpha1.Cluster) error {
 	return nil
 }
 
-//TODO: update git if template changes
+func (r *ReconcileCluster) updateTenantGitRepo(tenant types.NamespacedName, filename string) error {
+	tenantCR := &synv1alpha1.Tenant{}
+
+	err := r.client.Get(context.TODO(), tenant, tenantCR)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		} else {
+			return nil
+		}
+	}
+
+	if _, ok := tenantCR.Spec.GitRepoTemplate.TemplateFiles[filename]; !ok {
+		if tenantCR.Spec.GitRepoTemplate.TemplateFiles == nil {
```

Comment:
Can this even be `nil` if you just accessed it?

Reply:
Accessing a nil map doesn't panic, writing to one does:
https://play.golang.org/p/E77-4k1MPs-

That's a rather weird Golang quirk I found and caused nil pointer panics during testing.

Reply:
Hmm, weird indeed.
Then I'd suggest to untangle this nested if/else a little: Check if map is `nil` and instantiate an empty one if so. The other if can then be simpler.

Reply:
Untangled the nested IF statement.